src/cryptonote_core/tx_pool.cpp

```diff
@@ -917,26 +917,53 @@ namespace cryptonote
   {
     CRITICAL_REGION_LOCAL(m_transactions_lock);
     CRITICAL_REGION_LOCAL1(m_blockchain);
-    const relay_category category = include_sensitive ? relay_category::all : relay_category::broadcasted;
-    backlog.reserve(m_blockchain.get_txpool_tx_count(include_sensitive));
-    txpool_tx_meta_t tmp_meta;
-    m_blockchain.for_all_txpool_txes([this, &backlog, &tmp_meta](const crypto::hash &txid, const txpool_tx_meta_t &meta, const cryptonote::blobdata_ref *bd){
-      transaction tx;
-      if (!(meta.pruned ? parse_and_validate_tx_base_from_blob(*bd, tx) : parse_and_validate_tx_from_blob(*bd, tx)))
-      {
-        MERROR("Failed to parse tx from txpool");
-        // continue
-        return true;
-      }
-      tx.set_hash(txid);
-
-      tmp_meta = meta;
 
-      if (is_transaction_ready_to_go(tmp_meta, txid, *bd, tx))
-        backlog.push_back({txid, meta.weight, meta.fee});
+    std::vector<tx_block_template_backlog_entry> tmp;
+    uint64_t total_weight = 0;
 
+    // First get everything from the mempool, filter it later
+    m_blockchain.for_all_txpool_txes([&tmp, &total_weight](const crypto::hash &txid, const txpool_tx_meta_t &meta, const cryptonote::blobdata_ref*){
+      tmp.emplace_back(tx_block_template_backlog_entry{txid, meta.weight, meta.fee});
+      total_weight += meta.weight;
       return true;
-    }, true, category);
+    }, true, include_sensitive ? relay_category::all : relay_category::broadcasted);
+
+    // Limit backlog to 112.5% of current median weight. This is enough to mine a full block with the optimal block reward
```

Comment:
The max block weight is 2x the median weight, which can occur if fee pressure is high enough. I'd recommend 2.25x the median weight.

Reply:
Or you could use the block penalty algo to choose txs.

Reply:
I need to think about it more. The problem is that `is_transaction_ready_to_go` is a very slow function (it checks all RCT signatures), which is why this PR was created in the first place. So setting as low limit as reasonably possible is the goal here.

Reply:
Maybe this is a deeper architecture problem... txs in the mempool should ALWAYS be ready to go. If there are any conditions where a tx isn't ready to go, that implies a problem with how the mempool is being managed internally.

Reply:
Transactions can get invalid after a reorg, or even after a new block is received in case this block has conflicting transactions

Reply:
At minimum, a tx in the mempool should have valid proofs. The semantics could be invalid due to a hardfork (for example, changing the max output count), and the references could be invalid due to reorgs (ring member references and key images). Sadly, `check_tx_inputs()` is a do-everything function so you have to call it to check _anything_.

Reply:
The 'ideal' is out of scope for this PR though. Maybe an API endpoint for 'request more txs not in this tx set' would help you.
1. use the notifier to get an initial batch of txs
2. start mining
3. request small additional batches on a separate thread until you can't increase the size of your tx blob any more without incurring a penalty